mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1189,6 +1519,169 @@ struct Conv2DRewritePattern : public OpRewritePattern<T> {
     auto gemmC = b.create<miopen::TransformOp>(
         loc, transformedOutputMemRefType, op.output(), transformedOutputAttrs);
 
+    auto gemmC_Pad = gemmC;
+    bool isOutputPad = false;
+    // output padding start
+    if (convOpType == miopen::ConvOpType::Conv2DBwdWeightOpType &&
+        gemmK_extra > 0) {
```

Comment:
same as above, comment is needed to explain the general idea behind the algorithm.

also, matrix C is only padded in case of backward weight right now, not forward convolution. it needs to be documented.